### PR TITLE
align ML-KEM to FIPS 203 (final)

### PR DESCRIPTION
**Pull request by GiacomoPope**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,8 +35,8 @@ to learn about how Kyber works, and to try and create a clean, well commented
 implementation which people can learn from.
 
 This code is not constant time, or written to be performant. Rather, it was
-written so that the python code closely follows Algorithms 1-9 in the original
-[specification](https://pq-crystals.org/kyber/data/kyber-specification-round3-20210804.pdf).
+written so that the python code closely follows the Kyber specification
+[specification](https://pq-crystals.org/kyber/data/kyber-specification-round3-20210804.pdf) and [FIPS 203](https://csrc.nist.gov/pubs/fips/203/final). To cryptographic guarantees are made of this work.
```
Comment (tomato42) [inline README.md:39]: "Though no cryptographic guarantees are made in this work."?

File: src/kyber_py/ml_kem/ml_kem.py [hunk 2]
```diff
@@ -11,7 +12,7 @@
 class ML_KEM:
     def __init__(self, params):
         """
-        Initialise the ML-KEM with specified lattice parameters.
```
Comment (tomato42) [inline src/kyber_py/ml_kem/ml_kem.py:14]: PEP257 actually recommends dot at the end of the first line of the docstring...

File: tests/test_ml_kem.py [hunk 3]
```diff
@@ -90,80 +58,89 @@ def test_xof_failure(self):
...
+    consistency by generating key pairs
+    and shared secrets.
+    """
+
+    def generic_keygen_kat(self, ML_KEM, index):
+        with open("assets/ML-KEM-keyGen-FIPS203/internalProjection.json") as f:
+            data = json.load(f)
+        kat_data = data["testGroups"][index]["tests"]
+
+        for test in kat_data:
```
Comment (tomato42) [inline tests/test_ml_kem.py:99]: this again runs all the KATs as a single test... :disappointed:

Comment (GiacomoPope) [reply tests/test_ml_kem.py:99]: Yeah but there's only 70 in total so it doesn't really matter. I'm happy the test being "everything matches the test"

**Pull request merged**